sdk/core/azure-core/azure/core/pipeline/policies/_authentication.py

```diff
@@ -126,6 +126,8 @@ def send(self, request):
                 if "WWW-Authenticate" in response.http_response.headers:
                     request_authorized = self.on_challenge(request, response)
                     if request_authorized:
+                        # if we receive a challenge response, clear the 'insecure_domain_change' tag
+                        request.context.options.pop('insecure_domain_change', False)
```

Comment:
I'm not really a fan of the dependency between policies here.... Some SDKs have their own implementations of auth policies - that may not be aware of this flag.

Reply:
This hacky handler is only needed in the scenario where redirect happens and the new target uses a challenging auth (we never see such scenario before, but just to make some protection).

As far as I know, currently there is no custom SDK in such situation.